tests/Temporalio.Tests/Worker/WorkflowWorkerTests.cs

```diff
@@ -2363,7 +2343,7 @@ public class PostPatchWorkflow : PatchWorkflowBase
         }
     }
 
-    [Fact]
+    [Fact(Skip = "TODO(cretz): This is currently not throwing expected non-determinism errors in core")]
```

Comment:
@Sushisource - this is the test that, on the code below that runs:

```csharp
                var exc = await Assert.ThrowsAsync<WorkflowQueryFailedException>(
                    () => QueryWorkflowAsync(prePatchID));
```

This is not throwing the non-determinism error one might expect even with latest core. This is testing that using query to replay pre-patch code on a post-patch workflow will cause non-determinism (activity type change).

Reply:
OK, weird. I'll check this out - what's the easiest way for me to run just this test? I haven't learned much of the tooling for this yet.

Reply:
Download the [`dotnet` binary](https://dotnet.microsoft.com/en-us/download), clone this repo recursively, then in this dir run `dotnet build` (which runs Rust build, so you'll need `protoc` and such), remove the `Skip` message from the attribute, and then you can do something like:

    dotnet run --project tests/Temporalio.Tests -- -method "*.ExecuteWorkflowAsync_Patched_ProperlyHandled"

That should be enough for you to see. Feel free to comment out many parts of this multi-step test. Feel free to hit me up with any snags.

Reply:
Turns out this is because the version of server this SDK is using for tests doesn't support the sdk metadata and hence internal flags feature

Reply:
:+1: I will change the skip message to be more clear. I have opened https://github.com/temporalio/sdk-dotnet/issues/50. Semi-relatedly, I have also opened https://github.com/temporalio/sdk-java/issues/1746.